client/components/domains/map-domain-step/index.jsx

```diff
@@ -164,105 +164,21 @@ const MapDomainStep = React.createClass( {
 		this.recordEvent( 'formSubmit', this.state.searchQuery );
 		this.setState( { suggestion: null, notice: null } );
 
-		canMap( domain, error => {
-			if ( error ) {
-				this.handleValidationErrorMessage( domain, error );
-				return;
-			}
-
-			return this.props.onMapDomain( domain );
-		} );
-
 		canRegister( domain, ( error, result ) => {
 			if ( error ) {
+				if ( error.code === 'not_available_but_mappable' ) {
+					this.props.onMapDomain( domain );
+					return;
```

Comment:
Not sure if this is a rhetorical question or you did not read the code? `canRegister` is a poorly named call to the `is-available` endpoint, which confusingly returns three flags: `is_available`, `is_mappable` and `is_registrable`. Now, to make matters worse, these flags are converted into string errors and passed here in `error`... So, yeah, the code is more confusing than it should but we should untangle that one bit at a time.
But to make this part clearer, I've renamed the `canRegister` import to `checkDomainAvailability` - hope that make it clearer.
As to why it calls `onMapDomain` when a domain is registered - that's the prerequisite for mapping a domain. You can't map a domain that has not been registered.